### ISSUE TITLE
[bug] Indirect python_requires with version range from cache not found

### ISSUE THREAD

**Issue by fourbft** [conan-io/conan#13761]
### Environment details

* Operating System+version: Linux Ubuntu 22.04
* Conan version: 2.0.4
* Python version: 3.10.6

A python_requires package with a version range fails to be found in the local cache (even though a package matching the version range exists) when indirectly required through another package (via normal "requires"). 

Same scenario with a fixed version in the python_requires works.

### Steps to reproduce

Create a trivial package to serve as a python_requires:
```
$ mkdir pr
$ cd pr
$ conan new basic
```
Edit `conanfile.py`. Set package name to "pr". Create package in cache:
```
$ conan create .
```
Create a second package that uses the first as python_requires:
```
$ cd ..
$ mkdir use_pr
$ cd use_pr
$ conan new basic
```
Edit `conanfile.py`. Set package name to "use_pr". Add a line to the class header:
```
    python_requires = "pr/[>0]"
```
(Actually any version range, that includes the version of the "pr" package, will do.)
Create the package in cache:
```
$ conan create .
```
Package will be created without a problem (i. e. the pr/1.0 package will be found and used from cache).
Now create a third basic conan package:
```
$ cd ..
$ mkdir use_use_pr
$ cd use_use_pr
$ conan new basic
```
Edit `conanfile.py`. Set package name to e. g. `use_use_pr`. Add `use_pr/1.0` as normal requirement:
```
    def requirements(self):
        self.requires( "use_pr/1.0")
```
Try to create the package:
```
$ conan create .
```
Package creation will fail, because `pr`-package cannot be found, even though it is in the cache.

### Logs

```
$ conan create .
======== Exporting recipe to the cache ========
use_use_pr/1.0: Exporting package recipe: /home/user/tmp/20230424-conan2-bug-version_range_in_python_requires_when_in_local_cache/use_use_pr-package/conanfile.py
use_use_pr/1.0: Copied 1 '.py' file: conanfile.py
use_use_pr/1.0: Exported to cache folder: /home/user/.conan2/p/use_uc889bfa0cf41d/e
use_use_pr/1.0: Exported: use_use_pr/1.0#b1b720ecaeac4d8f07cde7a0c6369d75 (2023-04-24 18:05:46 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=RelWithDebInfo
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux

Profile build:
[settings]
arch=x86_64
build_type=RelWithDebInfo
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux


======== Computing dependency graph ========
pr/[>0]: Not found in local cache, looking in remotes...
pr/[>0]: Checking remote: conan-rme
Graph root
    cli
Requirements
    use_use_pr/1.0#b1b720ecaeac4d8f07cde7a0c6369d75 - Cache
*********************************************************
Recipe 'use_pr/1.0' seems broken.
It is possible that this recipe is not Conan 2.0 ready
If the recipe comes from ConanCenter check: https://conan.io/cci-v2.html
If it is your recipe, check if it is updated to 2.0
*********************************************************

ERROR: Package 'use_pr/1.0' not resolved: use_pr/1.0: Cannot load recipe.
Error loading conanfile at '/home/user/.conan2/p/use_p9732bd97ea5f3/e/conanfile.py': Cannot resolve python_requires 'pr/[>0]': Unable to find 'pr/[>0]' in remotes.
```

**Comment by memsharded** [comment]
Thanks very much for the clear way to reproduce this @fourbft, it really helped, I managed to reproduce, indeed a bug.
It will be fixed asap for next 2.0.5 patch release.

**Comment by fourbft** [comment]
Thanks for fixing so quickly, @memsharded ! Glad I could help.
